### ISSUE TITLE
'0 implementations' on type nested in fn

### ISSUE THREAD

**Issue by TyPR124** [rust-lang/rust-analyzer#3739]
I have the following:

```rust
#[derive(Clone)]
struct Foo;
fn bar() {
    #[derive(Clone)]
    struct Bar;
}
````

which shows '1 implementation' on `Foo` but '0 implementations' on `Bar`. I'm using version 0.2.100 as listed in VSCode.

**Comment by lnicola** [comment]
Might be a duplicate of #1559.

**Comment by lnicola** [comment]
Blocked on #1559.
